Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -677,6 +677,14 @@ ExceptionOr<void> ContainerNode::removeChild(Node& oldChild)
     rebuildSVGExtensionsElementsIfNecessary();
     dispatchSubtreeModifiedEvent();
 
+    auto* element = dynamicDowncast<Element>(oldChild);
```

Comment:
This is a wrong place to add this kind of logic. We need to have this code in `ContainerNode::removeAllChildrenWithScriptAssertion` and `ContainerNode::removeNodeWithScriptAssertion`, not this one function.